File: server/internal/models/flag.go
package models

type Flag struct {
	Iso2 string `json:"iso_2"`
	Name string `json:"name"`
}

type Characteristic struct {
	Name string `json:"name"`
	Type string `json:"type"`
}

type FlagCharacteristics struct {
	Iso2            string           `json:"iso_2"`
	Name            string           `json:"name"`
	Characteristics []Characteristic `json:"characteristics"`
}

type Characteristics struct {
	CharacteristicId string `json:"characteristicId"`
	Name             string `json:"name"`
	Type             string `json:"type"`
}

File: client/src/type-defs/flag.ts
export type SelectedFlags = (null | SelectedFlag)[][];

export type SelectedFlag = Flag & {
  playersTurn: number;
};

export type Characteristic = { type: string; name: string };

export type Flag = {
  iso_2: string;
  name: string;
  characteristics: Characteristic[];
};
